### PR TITLE
FOUNDATIONS: Retrieve Shopping Item By Id

### PR DESCRIPTION
**Pull request by tharaniannamalai**

### CODE REVIEW

File: ShoppingList.ConsoleApp/Models/ShoppingItems/Exceptions/NullArgumentShoppingItemException.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+using System;
+using Xeptions;
+
+namespace ShoppingList.ConsoleApp.Models.ShoppingItems.Exceptions
+{
+    public class NullArgumentShoppingItemException : Xeption
+    {
+        public NullArgumentShoppingItemException(Exception innerException)
+            : base(message: "Null argument shopping item error occured, fix the errors and try again.",
```
Comment (ShriHumrudha) [inline ShoppingList.ConsoleApp/Models/ShoppingItems/Exceptions/NullArgumentShoppingItemException.cs]: occurred typo, check in other instances too

File: ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Validations.RetrieveById.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        [Fact]
+        public void ShouldThrowValidationExceptionOnRetrieveByIdIfIdIsInvalidAndLogIt()
+        {
+            // given
+            int invalidId = default;
+            var invalidShoppingItemException = new InvalidShoppingItemException();
+
+            invalidShoppingItemException.AddData(
+                key: nameof(ShoppingItem.Id),
+                values: "Value is required.");
```
Comment (ShriHumrudha) [inline ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Validations.RetrieveById.cs:24]: Id is required ?

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: ShoppingList.ConsoleApp/Models/ShoppingItems/Exceptions/NullArgumentShoppingItemException.cs (hunk 1 above)
Comment (tharaniannamalai) [reply ShoppingList.ConsoleApp/Models/ShoppingItems/Exceptions/NullArgumentShoppingItemException.cs]: Comment Resolved

File: ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Validations.RetrieveById.cs (hunk 2 above)
Comment (tharaniannamalai) [reply ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Validations.RetrieveById.cs:24]: Value is required is a common message for id and quantity. Thanks.

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**